### ISSUE TITLE
cylc 7 gui uses a lot of suite CPU

### ISSUE THREAD

**Issue by matthewrmshin** [cylc/cylc#2256]
This is because it is making 8 HTTPS connections to the suite every second. Adding a print statement at the right place, I get:

```
('https://my.suite.host:43029/log/get_err_content?max_lines=10&prev_size=0', None, 'GET')
('https://my.suite.host:43029/state/get_summary_update_time', None, 'GET')
('https://my.suite.host:43029/state/get_state_summary', None, 'GET')
('https://my.suite.host:43029/state/get_state_summary', None, 'GET')
('https://my.suite.host:43029/info/get_first_parent_ancestors', None, 'GET')
('https://my.suite.host:43029/info/get_first_parent_ancestors?pruned=True', None, 'GET')
('https://my.suite.host:43029/info/get_first_parent_descendants', None, 'GET')
('https://my.suite.host:43029/info/get_all_families', None, 'GET')
```

We should reduced the number of connections. Possible solutions:

1. Consolidate all the calls to a smaller number, preferably a single connection only.
2. Introduce a new API method to allow the GUI to poll the suite for changes. Only invoke the other methods if there are changes to the suite.

**Comment by arjclark** [comment]
> 2. Introduce a new API method to allow the GUI to poll the suite for changes. Only invoke the other methods if there are changes to the suite

This sounds the preferable way to go so you do all your comms in one go + only retrieve differences from last known state, reducing data being transmitted round. The finesse would need to be in making sure that if a state change was missed, all the relevant changes since were still retrieved e.g. an interface of the form: `get_updates(last_known_state_tag)`

**Comment by dvalters** [comment]
@matthewrmshin where was the print statement placed to get this output?

**Comment by matthewrmshin** [comment]
Using a suite that looks like this:
```cfg
[scheduling]
    [[dependencies]]
        graph=t1 => t2
[runtime]
    [[t1]]
        script=false
    [[t2]]
        script=true
```

I got the following *user* times:

Test | 6.11.4 | 7.3.0
---- | ---- | ----
Control | 0.67 | 0.83
Mock GUI | 1.18 | 4.73

(where Control is to let the suite sits for 2 minutes doing nothing, and Mock GUI is a script that calls the same set of API calls as the GUI does for 100 times, with sleep 1 between each set.)

**Comment by arjclark** [comment]
Interesting that user time goes up for 7.3.0 control runs but I imagine that isn't a significant difference (though we kind expect 7 to be worse than 6.11.x based on Oliver's profiling).

**Comment by matthewrmshin** [comment]
(Sorry, some of these may also have been caused by #2239.)

**Comment by matthewrmshin** [comment]
Closed by #2271.
